From 6fd2552c9b2c99dc38ddc0880495caf10f264a9d Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 11 Aug 2012 18:10:28 +0200
Subject: [PATCH 098/227] MediaMon: Ensure to send status changed after disc
 mounted

If a USB key disc is inserted then a mount event is never triggered
and the media is never recognized.  If the disk is mounted before
mythfrontend is started or is mounted by an external agent then the
media is recognized.

This patch ensures that whenever a disk is mounted an event is triggered
and also adds some debug log output to show all media transitions.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/mythmedia.cpp |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythmedia.cpp b/mythtv/libs/libmythbase/mythmedia.cpp
index b089b38..3b744d0 100644
--- a/mythtv/libs/libmythbase/mythmedia.cpp
+++ b/mythtv/libs/libmythbase/mythmedia.cpp
@@ -159,7 +159,7 @@ bool MythMediaDevice::performMountCmd(bool DoMount)
                                            " find mounted media, but failed?");
                     return false;
                 }
-                m_Status = MEDIASTAT_MOUNTED;
+                setStatus(MEDIASTAT_MOUNTED); // emit statusChanged
                 onDeviceMounted();
                 LOG(VB_GENERAL, LOG_INFO,
                         QString("Detected MediaType ") + MediaTypeString());
@@ -472,6 +472,10 @@ MythMediaStatus MythMediaDevice::setStatus( MythMediaStatus NewStatus,
     // depending on the old and new status.
     if (NewStatus != OldStatus)
     {
+        LOG(VB_MEDIA, LOG_DEBUG,
+            QString("MythMediaDevice::setStatus %1 %2->%3")
+            .arg(getDevicePath()).arg(MediaStatusStrings[OldStatus])
+            .arg(MediaStatusStrings[NewStatus]));
         switch (NewStatus)
         {
             // the disk is not / should not be mounted.
-- 
1.7.9.5

